js/krakenfutures.js

```diff
@@ -0,0 +1,2048 @@
+                if (exchangeType === 'futures_inverse') {
+                    settle = base;
+                    settleId = baseId;
+                    inverse = true;
+                } else {
+                    settle = quote;
+                    settleId = quoteId;
+                    inverse = false;
+                }
+                linear = !inverse;
+                symbol = base + '/' + quote + ':' + settle;
+                if (future) {
+                    symbol = symbol + '-' + this.yymmdd (expiry);
+                }
+            }
+            result.push ({
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'settle': settle,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'settleId': settleId,
+                'type': type,
+                'spot': false,
+                'margin': false,
+                'swap': swap,
+                'future': future,
+                'option': false,
+                'index': index,
+                'active': undefined,
+                'contract': contract,
+                'linear': linear,
+                'inverse': inverse,
+                'contractSize': this.safeFloat (market, 'contractSize'),
+                'maintenanceMarginRate': undefined,
+                'expiry': expiry,
+                'expiryDatetime': this.iso8601 (expiry),
+                'strike': undefined,
+                'optionType': undefined,
+                'precision': {
+                    'amount': index ? undefined : this.parseNumber ('1'),
+                    'price': this.safeFloat (market, 'tickSize'),
+                },
+                'limits': {
+                    'leverage': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'amount': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'price': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                    'cost': {
+                        'min': undefined,
+                        'max': undefined,
+                    },
+                },
+                'info': market,
+            });
+        }
+        const settlementCurrencies = this.options['settlementCurrencies']['flex'];
+        const currencies = [];
+        for (let i = 0; i < settlementCurrencies.length; i++) {
+            const code = settlementCurrencies[i];
+            currencies.push ({
+                'id': code.toLowerCase (),
+                'numericId': undefined,
+                'code': code,
+                'precision': undefined,
+            });
+        }
+        this.currencies = this.deepExtend (currencies, this.currencies);
```

Comment:
This doesn't work how I want it to work, the currencies still aren't there when I run `krakenfutures currencies` 

We don't have a way to represent the same market with multiple settlement currencies, because it involves having multiple symbols with the same marketId, I've tried to set it up and they get overwritten. The compromise solution I've made is to represent the multi-collateral markets as `linear`, but this means that there are no markets with the currencies `[ 'USDT', 'GBP', 'EUR', 'USDC' ]` on `krakenfutures` so where I try to transfer `USDT` from `future` to `spot` I get `krakenfutures does not have currency code USDT`

I don't know what the best way to add these currencies to the exchanges currencies would be

Reply:
@kroitor @frosty00 Do you have any idea how I would solve this issue? I'm basically trying to add more currencies to `this.currencies` because there are some currencies allowed in multi-collateral trading, that aren't part of a market. It might be a good idea if we enable markets that have the same id, but different symbols, so that we can account for all the multicollateral markets on this, and also `bitfinex2`

Reply:
I'll try to address it as soon as I can.